addons/product/models/product_product.py

```diff
@@ -551,24 +552,41 @@ def get_display_name(name, code):
     def _search_display_name(self, operator, value):
         is_positive = operator not in expression.NEGATIVE_TERM_OPERATORS
         combine = expression.OR if is_positive else expression.AND
-        domains = [
-            [('name', operator, value)],
-            [('default_code', operator, value)],
-        ]
+        template_domains = [[('name', operator, value)]]
+        product_domains = [[('default_code', operator, value)]]
+        where_clause_data = [('product.template', 'id', template_domains), ('product.product', 'product_tmpl_id', product_domains)]
         if operator in ('=', 'in') or (operator.endswith('like') and is_positive):
             barcode_values = [value] if operator != 'in' else value
-            domains.append([('barcode', 'in', barcode_values)])
+            product_domains.append([('barcode', 'in', barcode_values)])
         if operator == '=' and isinstance(value, str) and (m := re.search(r'(\[(.*?)\])', value)):
-            domains.append([('default_code', '=', m.group(2))])
-        if partner_id := self.env.context.get('partner_id'):
+            product_domains.append([('default_code', '=', m.group(2))])
+        partner_id = self.env.context.get('partner_id')
+        if partner_id:
```

Comment:
why do you need to remove the walrus?
The variable is accessible where you use it, even if you use a walrus 🤔